finetune/base.py

```diff
@@ -397,13 +413,18 @@ def _cached_inference(self, Xs, predict_keys=None, n_examples=None):
         return predictions
 
     def _inference(self, Xs, predict_keys=None, n_examples=None):
+        if self.config.use_auxiliary_info:
+            context = Xs[1]
+            Xs = Xs[0]
+        else:
+            context = None
         Xs = self.input_pipeline._format_for_inference(Xs)
 
         if self._cached_predict:
-            return self._cached_inference(Xs=Xs, predict_keys=predict_keys, n_examples=n_examples)
+            return self._cached_inference(Xs=Xs, context=context, predict_keys=predict_keys, n_examples=n_examples)
         else:
-            input_fn = self.input_pipeline.get_predict_input_fn(Xs)
-            estimator, hooks = self.get_estimator(build_explain=PredictMode.EXPLAIN in predict_keys)
+            input_fn = self.input_pipeline.get_predict_input_fn(Xs, context=context)
+            estimator, hooks = self.get_estimator(build_explain=PredictMode.EXPLAIN in predict_keys, context_dim=self.input_pipeline.context_dim)
```

Comment:
If you already have a reference to self I don't think you really need to pass this through explicitly